latex/chain.tex

```diff
@@ -0,0 +1,787 @@
+      \\
+      \var{tail}';\eta = \var{seeds}
+      &
+      \eta' \leteq (\bseedn{bhb_{last}})\seedOp\eta
+    }
+    {
+      \left(
+        {\begin{array}{c}
+            \var{s_{now}} \\
+            \var{pps} \\
+            \var{sd} \\
+        \end{array}}
+      \right)
+      \vdash
+      {\left(\begin{array}{c}
+            \var{bhbs} \\
+            \var{seeds} \\
+      \end{array}\right)}
+      \trans{bhead}{\var{bh}}
+      {\left(\begin{array}{c}
+            \varUpdate{\var{bhbs};_w\var{bhb}} \\
+            \varUpdate{\var{seeds};_w \eta'} \\
+      \end{array}\right)}
+    }
+  \end{equation}
+  \caption{Block Header rules}
+  \label{fig:rules:block-header}
+\end{figure}
+
+\clearpage
+
+\subsection{New Epoch Transition}
+\label{sec:new-epoch-trans}
+
+\begin{figure}[ht]
+  \emph{New Epoch environments}
+  \begin{equation*}
+    \NewEpochEnv =
+    \left(
+      \begin{array}{r@{~\in~}lr}
+        \var{bhbs} & \seqof{\BHBody} & \text{sliding window of block header bodies} \\
+        \var{seeds} & \seqof{\Seed} & \text{sliding window of evolving seeds} \\
+        \var{pp_{new}} & \PParams & \text{new protocol parameters}\\
+      \end{array}
+    \right)
+  \end{equation*}
+  %
+  \emph{New Epoch states}
+  \begin{equation*}
+    \NewEpochState =
+    \left(
+      \begin{array}{r@{~\in~}lr}
+        \eta_0 & \Seed & \text{nonce at begining of epoch} \\
+        \var{blocks} & \BlocksMade & \text{blocks made in the epoch}\\
+        \var{es} & \EpochState & \text{epoch state} \\
+      \end{array}
+    \right)
+  \end{equation*}
+  %
+  \emph{New Epoch transitions}
+  \begin{equation*}
+    \_ \vdash \var{\_} \trans{NEWEPOCH}{\_} \var{\_} \subseteq
+    \powerset (\NewEpochEnv \times \NewEpochState \times \BHBody \times \NewEpochState)
+  \end{equation*}
+  \caption{New epoch transition-system types}
+  \label{fig:ts-types:new-epoch}
+\end{figure}
+
+\begin{figure}[ht]
+  \begin{equation}\label{eq:newepoch}
+    \inference[New-Epoch]
+    {
+      \var{last};\var{hs} \leteq bhbs
+      &
+      \var{e} \leteq \epoch{\bslot{bhb}}
+      &
+      \epoch{\bslot{last}} < e
+      \\
+      i \leteq \max\left\{j \mid
+      \bslot{(bhbs\downarrow j)} < (\firstSlot{e}) - 2\cdot (\fun{stableAfter}~\var{pps})\right\}
+      \\~\\
+      {
+        \left(
+        \begin{array}{l}
+          \var{e}\\
+          \var{pp_{new}}\\
+          \var{blocks}\\
+        \end{array}
+      \right)
+        \vdash \var{es} \trans{epoch}{} \var{es}
+      }
+    }
+    {
+      \left(
+        {\begin{array}{c}
+           \var{bhbs} \\
+           \var{seeds} \\
+           \var{pp_{new}} \\
+         \end{array}}
+      \right)
+      \vdash
+      \left(
+        {\begin{array}{c}
+            \eta_0 \\
+            \var{blocks} \\
+            \var{es} \\
+         \end{array}}
+      \right)
+    \trans{newepoch}{\var{bhb}}
+      \left(
+        {\begin{array}{c}
+            \varUpdate{\var{seeds}\downarrow i} \\
```

Comment:
Is this not a type error? `\var{seeds}\downarrow i` is a sequence of seeds, is it not?

Reply:
You're right, I shouldn't have used the down-arrow at all in this rule, I meant to be taking the j-th element of the list.  We can avoid dealing with indices all together if we pair the block header bodies with the seeds, but that comes with its own notational distractions...